gap/congruences/conguniv.gi

```diff
@@ -18,14 +18,43 @@ function(S)
   local fam, cong;
   fam := GeneralMappingsFamily(ElementsFamily(FamilyObj(S)),
                                ElementsFamily(FamilyObj(S)));
-  cong := Objectify(NewType(fam, IsUniversalSemigroupCongruence), rec());
+  cong := Objectify(NewType(fam,
+                            IsSemigroupCongruence and IsAttributeStoringRep),
+                    rec());
   SetSource(cong, S);
   SetRange(cong, S);
+  SetIsUniversalSemigroupCongruence(cong, true);
   return cong;
 end);
 
 #
 
+InstallMethod(IsUniversalSemigroupCongruence,
+"for a semigroup congruence",
+[IsSemigroupCongruence],
+function(cong)
+  return NrEquivalenceClasses(cong) = 1;
+end);
+
+#
+
+InstallMethod(IsUniversalSemigroupCongruence,
+"for a semigroup congruence",
+[IsRMSCongruenceByLinkedTriple],
+function(cong)
+  return cong!.n = UnderlyingSemigroup(Range(cong)) and
+         Length(cong!.colBlocks) = 1 and
```

Comment:
If I was being picky, I'd recommend having the semigroup equality check last, since it is the most expensive equality check here.